Content.Server/Actions/Spells/GiveItemSpell.cs

```diff
@@ -0,0 +1,50 @@
+using Content.Server.Hands.Components;
+using Content.Server.Items;
+using Content.Server.Notification;
+using Content.Shared.Actions.Behaviors;
+using Content.Shared.Actions.Components;
+using Content.Shared.Cooldown;
+using Content.Shared.Notification.Managers;
+using JetBrains.Annotations;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.Actions
+{
+    [UsedImplicitly]
+    [DataDefinition]
+    public class GiveItemSpell : IInstantAction
+    {
+        [ViewVariables] [DataField("castMessage")] public string CastMessage { get; set; } = "I CAST SPELL";
+        [ViewVariables] [DataField("coolDown")] public float CoolDown { get; set; } = 1f;
+        [ViewVariables] [DataField("spellItem")] public string ItemProto { get; set; } = default!;
+
+        [ViewVariables] [DataField("castSound")] public string? CastSound { get; set; } = default!;
+
+        //Rubber-band snapping items into player's hands, originally was a workaround, later found it works quite well with stuns
+        //Not sure if needs fixing
+
+        public void DoInstantAction(InstantActionEventArgs args)
+        {
+            if (!args.Performer.TryGetComponent<SharedActionsComponent>(out var actions)) return;
+            var caster = args.Performer;
+            var casterCoords = caster.Transform.Coordinates;
+            var spawnedProto = caster.EntityManager.SpawnEntity(ItemProto, casterCoords);
+            caster.PopupMessageEveryone(CastMessage);
+            args.PerformerActions?.Cooldown(args.ActionType, Cooldowns.SecondsFromNow(CoolDown));
+            if (!caster.TryGetComponent<HandsComponent>(out var hands))
+            {
+                caster.PopupMessage("You don't have hands!");
+                return;
+            }
+            caster.GetComponent<HandsComponent>().PutInHandOrDrop(spawnedProto.GetComponent<ItemComponent>());
+            if (CastSound != null)
+            {
+                SoundSystem.Play(Filter.Pvs(caster), CastSound, caster);
+            }
+            else return;
+        }
```

Comment:
1. Not sure how to fix that
2. I halt code execution if I do not get it
3. fixed
4. validation fixed, not sure if spell should be used with no hands as it would render it pretty much useless
5. I halt code execution 
6. Not sure about the implementation
7. Shouldn't it? The spell isn't intended to support non-item entities, shouldn't it throw by design?
8. fixed
9. fixed

Reply:
1. Fixed
2. Fixed
3. Should be fixed?
4. Fixed
5. Should be fixed?